custom_dataset.py

```diff
@@ -0,0 +1,408 @@
+# Copyright (C) 2020 and later: Google, Inc.
+
+import configparser
+import functools
+import numpy as np
+import os
+import pathlib
+import tensorflow as tf
+import tensorflow_addons as tfa
+
+
+# AUTOTUNE allows TensorFlow to find a good allocation of CPU budget for
+# performance optimization
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+
+class DatasetBuilder:
+    def __init__(self, config_path='configs/sample_config.ini', one_hot=True):
+        """Read and set configuration from config file (.ini file) and create
+        tf.Dataset object or input function according to configuration.
+
+        Args:
+            config_path: Str, path to config (.ini) file.
+            one_hot: Bool, whether or not to return label as one-hot encoding.
+
+        Raises:
+            ValueError: if values in config file does not have the correct type.
+        """
+        # Set one-hot encoding setting
+        self.ONE_HOT = one_hot
+
+        # Parse config file
+        config = configparser.ConfigParser()
+        config.read(config_path)
+
+        # Get (and check) configuration
+        # Get dataset info
+        self._TRAIN_DATA_DIR = config.get('DATASET', 'TRAIN_DATA_DIR')
+        self._TEST_DATA_DIR = config.get('DATASET', 'TEST_DATA_DIR')
+        self._LABEL_FILE = config.get('DATASET', 'LABEL_FILE')
+
+        # Get image info
+        self._HEIGHT = config.getint('IMAGE', 'HEIGHT')
+        self._WIDTH = config.getint('IMAGE', 'WIDTH')
+        self._GRAYSCALE_IN = config.getboolean('IMAGE', 'GRAYSCALE_IN')
+        self._GRAYSCALE_OUT = config.getboolean('IMAGE', 'GRAYSCALE_OUT')
+
+        # Get traning and testing spec
+        self._TRAIN_BATCH_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                              'TRAIN_BATCH_SIZE')
+        self._TEST_BATCH_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                             'TEST_BATCH_SIZE')
+        self._SHUFFLE_BUFFER_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                                 'SHUFFLE_BUFFER_SIZE')
+        self._PREFETCH_BUFFER_SIZE = config.getint('TRAIN_TEST_SPEC',
+                                                  'PREFETCH_BUFFER_SIZE')
+
+        # Get data augmentation spec
+        self._RANDOM_ROTATE = config.getboolean('DATA_AUG', 'RANDOM_ROTATE')
+        self._ROTATE_STDDEV = config.getfloat('DATA_AUG', 'ROTATE_STDDEV')
+
+        self._RANDOM_ZOOM = config.getboolean('DATA_AUG', 'RANDOM_ZOOM')
+        self._ZOOM_PERCENT = config.getfloat('DATA_AUG', 'ZOOM_PERCENT')
+        self._ZOOM_STDDEV = config.getfloat('DATA_AUG', 'ZOOM_STDDEV')
+
+        self._RESIZE = config.getboolean('DATA_AUG', 'RESIZE')
+        self._RESIZE_HEIGHT = config.getint('DATA_AUG', 'RESIZE_HEIGHT')
+        self._RESIZE_WIDTH = config.getint('DATA_AUG', 'RESIZE_WIDTH')
+
+        # Label conversion
+        self._CLASS_NAMES = [] # Class names in the same order as in label file
```

Comment:
Just
self._CLASS_NAMES = [s.strip() for s in open(self._LABEL_FILE).readlines()]
instead of the following 6 lines.